mentor/messages.py

```diff
@@ -0,0 +1,370 @@
+
+
+# u('string') replaces the forwards-incompatible u'string'
+if six.PY3:
+    def u(string):
+        return string
+else:
+    import codecs
+
+    def u(string):
+        return codecs.unicode_escape_decode(string)[0]
+
+# dict.iteritems(), dict.iterkeys() is also incompatible
+if six.PY3:
+    iteritems = dict.items
+    iterkeys = dict.keys
+else:
+    iteritems = dict.iteritems
+    iterkeys = dict.iterkeys
+
+class Message(dict):
+    """ A dictionary that provides attribute-style access.
+    """
+
+    def __contains__(self, k):
+        """
+        """
+        try:
+            return dict.__contains__(self, k) or hasattr(self, k)
+        except:
+            return False
+
+    # only called if k not found in normal places
+    def __getattr__(self, k):
+        """ Gets key if it exists, otherwise throws AttributeError.
+        """
+        try:
+            # Throws exception if not in prototype chain
+            return object.__getattribute__(self, k)
+        except AttributeError:
+            try:
+                return self[k]
+            except KeyError:
+                raise AttributeError(k)
+
+    def __setattr__(self, k, v):
+        """
+        """
+        try:
+            # Throws exception if not in prototype chain
+            object.__getattribute__(self, k)
+        except AttributeError:
+            try:
+                self[k] = v
+            except:
+                raise AttributeError(k)
+        else:
+            object.__setattr__(self, k, v)
+
+    def __delattr__(self, k):
+        """ Deletes attribute k if it exists, otherwise deletes key k. A KeyError
+            raised by deleting the key--such as when the key is missing--will
+            propagate as an AttributeError instead.
+        """
+        try:
+            # Throws exception if not in prototype chain
+            object.__getattribute__(self, k)
+        except AttributeError:
+            try:
+                del self[k]
+            except KeyError:
+                raise AttributeError(k)
+        else:
+            object.__delattr__(self, k)
+
+
+    def __repr__(self):
+        """ Invertible* string-form of a Message.
+        """
+        keys = list(iterkeys(self))
+        keys.sort()
+        args = ', '.join(['%s=%r' % (key, self[key]) for key in keys])
+        return '%s(%s)' % (self.__class__.__name__, args)
+
+    def __dir__(self):
+        return self.keys()
+
+def Cpus(value):
+    return Message({'name': 'cpus', 'role': '*', 'scalar': Message({'value':value}), 'type': 'SCALAR'})
+
+def Mem(value):
+    return Message({'name': 'mem', 'role': '*', 'scalar':  Message({'value':value}), 'type': 'SCALAR'})
+
+def Disk(value):
+    return Message({'name': 'disk', 'role': '*', 'scalar': Message({'value':value}), 'type': 'SCALAR'})
+
+def Ports(begin,end):
+    return Message({'name': 'disk', 'role': '*', 'ranges': Message({'range':[Message({'begin': begin, 'end': end})]}), 'type': 'SCALAR'})
```

Comment:
God yes, maybe setup autopep8 or something, I did not do anything on this package